src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -12,76 +11,91 @@ internal sealed partial class LocalRewriter
     {
         public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstructionAssignmentOperator node)
         {
+            Debug.Assert(node.DeconstructSteps != null);
+            Debug.Assert(node.AssignmentSteps != null);
+
             var temps = ArrayBuilder<LocalSymbol>.GetInstance();
             var stores = ArrayBuilder<BoundExpression>.GetInstance();
+            var placeholders = ArrayBuilder<BoundValuePlaceholderBase>.GetInstance();
 
-            var lhsReceivers = ArrayBuilder<BoundExpression>.GetInstance();
-            foreach (var variable in node.LeftVariables)
-            {
-                // This will be filled in with the LHS that uses temporaries to prevent
-                // double-evaluation of side effects.
-                lhsReceivers.Add(TransformCompoundAssignmentLHS(variable, stores, temps, isDynamicAssignment: false));
-            }
+            // evaluate left-hand-side side-effects
+            var lhsTemps = LhsSideEffects(node.LeftVariables, temps, stores);
 
+            // get or make right-hand-side values
             BoundExpression loweredRight = VisitExpression(node.Right);
-            ImmutableArray<BoundExpression> rhsValues;
+            AddPlaceholderReplacement(node.DeconstructSteps[0].TargetPlaceholder, loweredRight);
+            placeholders.Add(node.DeconstructSteps[0].TargetPlaceholder);
```

Comment:
I'm not sure I understand the question.
Here we bootstrap the first placeholder for the RHS value. After the RHS value is deconstructed, it will populate proper placeholders for subsequence deconstructs, and so on. Those same placeholders are also referenced by the assignment step.
At the very end, we clear all those "carried-through" placeholders.